indi-atik/atik_ccd.cpp

```diff
@@ -56,34 +58,47 @@ void ATIK_CCD_ISInit()
         iAvailableCamerasCount = 0;
         std::vector<std::string> cameraNames;
 
-        iAvailableCamerasCount = ArtemisDeviceCount();
-        if (iAvailableCamerasCount > MAX_DEVICES)
-            iAvailableCamerasCount = MAX_DEVICES;
+        IDLog("Atik Cameras API V%d DLL V%d initializing.", ArtemisAPIVersion(), ArtemisDLLVersion());
+
+        for (int loop = 0; loop < MAX_CONNECTION_RETRIES; loop++)
+        {
+            iAvailableCamerasCount = ArtemisDeviceCount();
+
+            if (0 < iAvailableCamerasCount)
+                break;
+
+            IDLog("No Atik Cameras detected on attempt %d/%d, retrying...", loop+1, MAX_CONNECTION_RETRIES);
+        }
+
         if (iAvailableCamerasCount <= 0)
-            IDLog("No Atik Cameras detected. Power on?");
-        else
         {
-            for (int i = 0; i < iAvailableCamerasCount; i++)
```

Comment:
Is this correct? You should be looping among available atik devices not just cameras.
It is always the case that number of atik devices >= numer of atik cameras

Reply:
In the context of indi_atik_ccd, as far as I understood the code, only cameras are filtered in. Filter wheels are managed by the other driver.

Reply:
Now I see it. It is iAvailableCamerasCount variable name that is confusing. It is assigned value from ArtemisDeviceCount() so in fact it should be called iAvailableDevicesCount.

Reply:
I'll change this, good point.

Reply:
Actually, I think the driver has a bug here: the number of cameras in `cameraNames` does not include the other devices, and connecting for instance a camera and a filter wheel will cause an index overflow...

Reply:
I use both Atik CCD and EFW and have not observed this in the past.

Reply:
If you connect a CCD and an EFW, `iAvailableCamerasCount` will equal 2, but only one of `cameras` cells will be uninitialized (depending on which device is enumerated first), and `cameraNames` will contain only one element. I don't really understand why the loops below are not crashing when de-referencing `cameras` content. Unless I'm missing something?

Reply:
I chose to add safeguards on the `cameras` element considered.